cmd/bnsd/x/blueaccount/handler.go

```diff
@@ -0,0 +1,505 @@
+package blueaccount
+
+import (
+	"regexp"
+
+	weave "github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/gconf"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+func RegisterQuery(qr weave.QueryRouter) {
+	NewDomainBucket().Register("domain", qr)
+	NewAccountBucket().Register("account", qr)
+}
+
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	r = migration.SchemaMigratingRegistry("blueaccount", r)
+
+	domains := NewDomainBucket()
+	accounts := NewAccountBucket()
+
+	r.Handle(&RegisterDomainMsg{}, &registerDomainHandler{
```

Comment:
Just a question: In the previous extension implementation every handler created its own bucket and accessed it indiviually. In this implementation, buckets are shared. Is there a possibility that this could case side effects or some unexpected result?

Reply:
No, this is ok. Bucket is a proxy for accessing the store. Each handler has complete access to the store.

There are more implementations that do this. For example every extension that relies on `cash.Controller` is moving tokens while modifying that extension entity.
Look also at `gov`. Some handlers there keep reference to 3 buckets :dancer: